lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
I am sorry for adding that `+`, I was looking for `HTTP message header`, not `HTTP header`, my bad.

So, you prefer:
```
def strip_header_continuation_from_ssl_cert(ssl_client_cert)
  cert_header = "-----BEGIN CERTIFICATE-----\n"
  cert_footer = "\n-----END CERTIFICATE-----"
  cert_header + ssl_client_cert.split(/[ |\t]/)[2..-3].join("\n") + cert_footer
end
```
or just:
```
  "-----BEGIN CERTIFICATE-----\n" + ssl_client_cert.split(/[ |\t]/)[2..-3].join("\n") + "\n-----END CERTIFICATE-----"
```
?
I prefer first version, because second one is a bit to long for me. 

About test. I never wrote ruby tests before, but I will try to write it.